samza-sql/src/main/java/org/apache/samza/sql/translator/JoinTranslator.java

```diff
@@ -114,41 +122,51 @@ void translate(final LogicalJoin join, final TranslatorContext context) {
       log.info(streamFieldNames.get(streamKeyIds.get(i)) + " with " + tableFieldNames.get(tableKeyIds.get(i)));
     }
 
-    SamzaSqlRelMessageJoinFunction joinFn =
-        new SamzaSqlRelMessageJoinFunction(join.getJoinType(), isTablePosOnRight, streamKeyIds, streamFieldNames,
-            tableKeyIds, tableFieldNames);
-
-    SamzaSqlRelRecordSerdeFactory.SamzaSqlRelRecordSerde keySerde =
-        (SamzaSqlRelRecordSerdeFactory.SamzaSqlRelRecordSerde) new SamzaSqlRelRecordSerdeFactory().getSerde(null, null);
-    SamzaSqlRelMessageSerdeFactory.SamzaSqlRelMessageSerde valueSerde =
-        (SamzaSqlRelMessageSerdeFactory.SamzaSqlRelMessageSerde) new SamzaSqlRelMessageSerdeFactory().getSerde(null, null);
-
-    // Always re-partition the messages from the input stream by the composite key and then join the messages
-    // with the table. For the composite key, provide the corresponding table names in the key instead of using
-    // the names from the stream as the lookup needs to be done based on what is stored in the local table.
-    MessageStream<SamzaSqlRelMessage> outputStream =
-        inputStream
-            .partitionBy(m -> createSamzaSqlCompositeKey(m, streamKeyIds,
-                getSamzaSqlCompositeKeyFieldNames(tableFieldNames, tableKeyIds)),
-                m -> m,
-                KVSerde.of(keySerde, valueSerde),
-                intermediateStreamPrefix + "stream_" + joinId)
-            .map(KV::getValue)
-            .join(table, joinFn);
+    MessageStream<SamzaSqlRelMessage> outputStream;
+
+    if (isRemoteTable) {
+      List<String> tableNameParts = (isTablePosOnRight ? join.getRight() : join.getLeft())
+          .getTable()
+          .getQualifiedName();
+      String remoteTableName = SqlIOConfig.getSourceFromSourceParts(tableNameParts);
```

Comment:
Do we need metrics for the RemoteTableJoins?

Reply:
Yes - we will definitely need to have metrics similar to what Shenoda is adding. I will add them as part of a separate effort, once Shenoda commits his changes for filter and projection.

Reply:
Can you open a Jira to track this as well as the throttling and retry work?

Reply:
Sure